### PR TITLE
[#175183261] Automatically add story_id to commit message

### PR DESCRIPTION
**Pull request by debiff**
## Short Description
This PR automatically prepend the story_id to the commit message by extracting it from the branch name if it is in the form: **`story_id`**`-description`

### CODE REVIEW

**Comment by balanza** [comment]
What happens if the branch name is in an arbitrary form? Like `my-fancy-branch` for example. 
I think the resulting commit message would be `[#] commit message`. (I didn't try it actually).

**Comment by debiff** [comment]
Yes, you're right. It's not the expected behaviour (the message should results in `[#my] commit message`).

Check that `STORY_ID` is a number can be a solution? It not ensure that it's effectively an existing "Pivotal" story but it can fix the issue.

**Comment by balanza** [comment]
I think `STORY_ID` will be enough. The point is to keep things working even for those projects that don't have a strict convention over branch/commit naming.

File: hooks/prepare-commit-msg [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+TOTAL_MSG_LENGHT=$((${#COMMIT_MSG}+${#STORY_ID}+3))
+
+if (( $TOTAL_MSG_LENGHT < $MAX_MSG_LENGHT)); then
+   if [[ $STORY_ID =~ $IS_A_NUMBER_REGEX ]] ; then
+      sed -i '.bak' "1s/^/[#$STORY_ID] /" $1
+   fi
+else
+   echo "Commit message longer than ${MAX_MSG_LENGHT} characters"
+   exit 1
+fi
```
Comment (balanza) [inline hooks/prepare-commit-msg]: Should we invert the check? This way we apply no changes if we found no story point (it's the case we have no constrained logic on messages).

```suggestion
if [[ $STORY_ID =~ $IS_A_NUMBER_REGEX ]] ; then
   if (( $TOTAL_MSG_LENGHT < $MAX_MSG_LENGHT)); then
      sed -i '.bak' "1s/^/[#$STORY_ID] /" $1
   else
      echo "Commit message longer than ${MAX_MSG_LENGHT} characters"
      exit 1
   fi
fi 
```

Comment (debiff) [reply hooks/prepare-commit-msg]: Yes, you're right! Fixed!

**Pull request merged**